src/libfsm/fsm.c

```diff
@@ -82,6 +82,12 @@ fsm_free(struct fsm *fsm)
 	free(fsm);
 }
 
+const struct fsm_options *
+fsm_getoptions(const struct fsm *fsm)
+{
+	return fsm->opt;
+}
+
```

Comment:
Would `fsm_setoptions()` suit you, after clone?
There are a few operations which call `fsm_new()`, and it'd be odd to provide a `*_with_opts()` variant for all of them.